src/runtime_src/driver/zynq/drm/zocl/sched_exec.c

```diff
@@ -1246,8 +1300,23 @@ static int
 add_cmd(struct sched_cmd *cmd)
 {
 	int ret = 0;
+	int num_masks = 0;
+	int mask_idx = 0;
 
 	SCHED_DEBUG("-> add_cmd\n");
+
+	/*
+	 * We always check if given cmd is set wrong CU mask.
+	 */
+	num_masks = cu_masks(cmd);
+	if (opcode(cmd) == ERT_START_CU) {
+		for (mask_idx = 0; mask_idx < num_masks; mask_idx++) {
+			if (cmd->packet->data[mask_idx] &
+			    cmd->exec->cu_valid[mask_idx])
```

Comment:
My suggestion is not doing this here since you are toching global variable exec->cu_valid[] without holding any lock, Also the application has to change to match exactly the valid CU to trigger a run

Reply:
Agree, after discussed with Min, we could just remove this check for now.

Reply:
Agree. Since today OpenCL layer will make sure the CU_MASK is correct.